### PR TITLE
Chore: Migrate BaseKYCWall.js to function component

### PR DESCRIPTION
**Pull request by teneeto**
### Details

This change is made to migrate the [src/components/KYCWall/BaseKYCWall.js](https://github.com/Expensify/App/blob/main/src/components/KYCWall/BaseKYCWall.js) from class to a function component.

Note: This PR only addresses a reverted code of a previously [merged PR](https://github.com/Expensify/App/pull/28798)

### Fixed Issues

$: [16170](https://github.com/Expensify/App/issues/16170)
PROPOSAL:  [16170](https://github.com/Expensify/App/issues/16170)

### Tests

-  log into the app
-  press the FAB and request money from e.g. workspace that you are a part of (to make it easier and not have to jump between accounts)
-  select the said workspace and press on pay with Expensify
-  confirm that the pop up shows in which you can select an option and be navigated to it

- [x] Verify that no errors appear in the JS console

### Offline tests

-  log into the app
-  press the FAB and request money from e.g. workspace that you are a part of (to make it easier and not have to jump between accounts)
-  select the said workspace and press on pay with Expensify
-  confirm that the pop up shows in which you can select an option and be navigated to it

### QA Steps

-  log into the app
-  press the FAB and request money from e.g. workspace that you are a part of (to make it easier and not have to jump between accounts)
-  select the said workspace and press on pay with Expensify
-  confirm that the pop up shows in which you can select an option and be navigated to it

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
        - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

273612528-d41515a9-0469-414f-a13b-a5c497c4f7d7

</details>

<details>
<summary>Mobile Web - Chrome</summary>

Screenshot 2023-12-01 at 14 50 56

</details>

<details>
<summary>Mobile Web - Safari</summary>

Screenshot 2023-12-01 at 16 44 53

</details>

<details>
<summary>Desktop</summary>

Screenshot 2023-12-01 at 14 54 18

</details>

<details>
<summary>iOS</summary>

273612698-67718998-b9bb-4c7f-a8da-7c9f970aa715

</details>

<details>
<summary>Android</summary>

Screenshot 2023-12-01 at 15 54 47

</details>

### CODE REVIEW

File: src/components/KYCWall/BaseKYCWall.js [hunk 1]
```diff
@@ -15,90 +15,112 @@ import CONST from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
 import {defaultProps, propTypes} from './kycWallPropTypes';
 
+const POPOVER_MENU_ANCHOR_POSITION_HORIZONTAL_OFFSET = 20;
```
Comment (rezkiy37) [inline src/components/KYCWall/BaseKYCWall.js:18]: I would suggest to add a comment to describe why the const exists.

**Review by rezkiy37 — COMMENTED** [review]
Left a comment, besides LGTM.

File: src/components/KYCWall/BaseKYCWall.js (hunk 1 above)
Comment (teneeto) [reply src/components/KYCWall/BaseKYCWall.js:18]: This was already there and was requested to be moved to a CONST in Previous discussion. Can't tell exactly why it's added. I'm only fixing a refactor issue. https://github.com/Expensify/App/pull/28798#discussion_r1345650106

**Review by bondydaa — DISMISSED** [review]

**Comment by bondydaa** [comment]
oh @abdulrahuman5196 can you handle checklists?

File: src/libs/actions/PaymentMethods.ts [hunk 2]
```diff
@@ -214,7 +214,7 @@ function clearDebitCardFormErrorAndSubmit() {
     Onyx.set(ONYXKEYS.FORMS.ADD_DEBIT_CARD_FORM, {
         isLoading: false,
         errors: undefined,
-        setupComplete: false,
```
Comment (abdulrahuman5196) [inline src/libs/actions/PaymentMethods.ts:217]: @teneeto Why are we changing this to true?

Comment (teneeto) [reply src/libs/actions/PaymentMethods.ts:217]: apologies @abdulrahuman5196 just seeing this, let me have a look.

Comment (teneeto) [reply src/libs/actions/PaymentMethods.ts]: @abdulrahuman5196, I've reverted that change, that was a wrong code update. Thanks for Spotting it.

**Comment by teneeto** [comment]
@bondydaa @abdulrahuman5196, made code changes, let's have a look at it again.

**Comment by abdulrahuman5196** [comment]
Thank you. Will check again in my morning.

**Comment by abdulrahuman5196** [comment]
Reviewing now

**Comment by abdulrahuman5196** [comment]
@teneeto The author's checklist only has 2 platform videos?

**Review by abdulrahuman5196 — APPROVED** [review]
Changes looks good and works well. Reviewers checklist is also complete.

All yours. @bondydaa 

🎀 👀 🎀
C+ Reviewed

**Comment by abdulrahuman5196** [comment]
@teneeto To fix author's checklist as mentioned here https://github.com/Expensify/App/pull/31538#issuecomment-1836100929

**Comment by teneeto** [comment]
> @teneeto To fix author's checklist as mentioned here [#31538 (comment)](https://github.com/Expensify/App/pull/31538#issuecomment-1836100929)

Hi @abdulrahuman5196, added all screenshots and updated tests steps too

**Review by bondydaa — APPROVED** [review]
Just want to confirm, we tested that this works and doesn't cause a crash when you try to sign in, right?

**Pull request merged**